SendLog.py

```diff
@@ -126,90 +220,576 @@ def event_stream(xcos_file_id):
+	     w.write("0\n0\n0\n0\n0\n0\n0\n0\n0\n0") # initialize all tktime values to 0
+	     w.close()
+	  
+	     for i in range(10):
+	     	open("values/"+Details.uid+"_tk"+str(i+1)+".txt","w").close(); # create empty tk text files
+	     return 0
+	elif line=="Stop": # at last the val.txt contains "Stop" indicating the ending process
+	     return 1
+	return 2
+
+# function which changes flaoting to req scientific format    
+def changeFormat(n):
+	n = float(n)
+	check = 1
+	exp = 0
+	if n<0:
+		n = n*-1
+		check=-1
+	while not(0<=n and n<1):
+		n = n/10
+		exp=exp+1
+	n=n*check
+	n = "%.3f" % n
+	exp = "%02d" % (exp,)
+	
+	formated = n+"E+"+exp
+	
+	return formated
+	
+# function which appends the 'updated' ('new') value to the file    # '=> Dattatreya <=' 
+def AppendtoTKfile(fname,tlist,i):
+	tl = tlist.split('  ') # converting string to list which stores the Period parameter at 0 index, data to be appended at index 1
+	
+	if(tl[1]==''):
+		return # if data is empty, do not append it
+	with open("values/"+Details.uid+"_tktime.txt", 'r') as file:
+    		tktime = file.readlines() # read the tktime folder to get tkfile time at respective index (tk1.txt at 0 index)
+	time = float(tktime[i])+float(tl[0]) # update the time
+	tktime[i]=str(time)+'\n'
+
+	with open("values/"+Details.uid+"_tktime.txt", 'w') as file:
+    		file.writelines( tktime ) # write the updated the time to tktime.txt
+	
+	data = tl[1]
+	time = changeFormat(time)
+	
+	line = time+"  "+data+"\n"
+	
+	w= open(fname,'a') # append data to the tk.txt
+	w.write(line)
+	w.close()
+	
+
+# function which take values from val.txt send the data to  append them in their respective 'tk' files.        # '=> Dattatreya <=' 
+def getDetails():
+	r=open("values/"+Details.uid+"_val.txt","r")
+	line=r.readline()
+	tklist=line.split(',')
+	
+	for i in range(len(tklist)):
+		
+		tl = tklist[i].split('  ')
+		
+		if  len(tklist)==1 or len(tl)==1:
+			break
+		else:
+			fname="values/"+Details.uid+"_tk"+str(i+1)+".txt"
+
+			AppendtoTKfile(fname,tklist[i],i)
+	r.close()
+# function which makes the initialisation of thread    # '=> Dattatreya <=' 
+def getDetailsThread():
+	if Details.tkbool:
+		getDetails()
+		import threading 
+		from threading import Timer
+		threading.Timer(0.1,getDetailsThread).start()  # calls the same function adter 0.1 second
+
+def stopDetailsThread():     # '=> Dattatreya <=' 
+	Details.tkbool = False # stops the thread
+	import os, glob
+	for filename in glob.glob("values/"+Details.uid+"*"):
+    		os.remove(filename) # deletes all files created under the 'uid' name
+
 
 # Route that will process the file upload
 @app.route('/upload', methods=['POST'])
 def upload():
+    # Get the file
+    print( "upload")
+    file = request.files['file']
+    #flags to check if both TOWS_c and FROMWSB are present
+    flag1=0
+    flag2=0
```

Comment:
Please use a relevant variable name. What is the purpose of the ```flag``` maybe that could be used as a variable name